xontrib/uncpushd.py

```diff
@@ -0,0 +1,146 @@
+# And, if you must proceed , %WINDIR% is probably the worst fallback to use!
+# Either (ordinary) user will fail reading or trying and failing to write a file, or (privileged) user may succeed in writing, possibly clobbering something important.
+# -- there.  I feel much better.  To proceed...)
+
+import argparse
+
+from xonsh.lazyasd import LazyObject
+from xonsh.built_ins import subproc_captured_object
+from xonsh.platform import ON_WINDOWS
+
+_uncpushd_choices = dict(enable=1, disable=0, show=None)
+
+def _uncpushd_parser():
+    parser = argparse.ArgumentParser(prog="uncpushd", description='Enable or disable CMD.EXE check for UNC path.')
+    parser.add_argument('action', choices=_uncpushd_choices, default='show')
+    return parser
+
+uncpushd_parser = LazyObject(_uncpushd_parser, globals(), 'uncpushd_parser')
+
+del _uncpushd_parser
+
+def uncpushd(args=None, stdin=None):
+    """Set, Clear or display current value for DisableUNCCheck in registry, which controls
+    whether CMD.EXE complains when working directory set to a UNC path.
+
+    In new windows install, value is not set, so if we cannot query the current value, assume check is enabled
+    (meaning CMD.EXE complains).
+
+    Does nothing on non-Windows platforms.
+    """
+    if not ON_WINDOWS:
+        return None, None, 0
+
+    try:
+        args = uncpushd_parser.parse_args(args)
+    except SystemExit:
+        return None, None
+
+    if _uncpushd_choices[args.action] is None:             # show current value
+        co = subproc_captured_object(
+            ['reg', 'query', '"hkcu\software\microsoft\command processor"', '-v', 'DisableUNCCheck'])
+        if co.returncode == 0:
+            return 'enabled\n' if co.stdout[-3:-2] == '1' else 'disabled\n'\
+                , None, 0-int(co.stdout[-3:-2])
+        else:
+            return 'disabled\n', None, 0
+    else:                               # set to 1 or 0
+        co = subproc_captured_object(
+            ['reg', 'add', '"hkcu\software\microsoft\command processor"', '-v', 'DisableUNCCheck', '-t', 'REG_DWORD',
+             '-d', '1' if _uncpushd_choices[args.action] else '0', '-f'])
+        return None if co.returncode == 0 else co.stdout, None if co.returncode == 0 else co.stderr, co.returncode
+    pass
+
+import builtins
+import os
+from xonsh.dirstack import pushd, popd, DIRSTACK
+import subprocess
+
+def _do_subproc(args, msg=''):
+    """Because `subproc_captured_object` fails with error `Workstation Service not started` on a `NET USE dd: \\localhost\share`..."""
+    co = subprocess.run(args, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    # if co.returncode != 0:
+    #     print('subproc returncode {}, msg: {}'.format(co.returncode, msg))
+    #     print(' '.join(co.args))
+    #     print(co.stdout)
+    #     print(co.stderr)
+    #     ##assert co.returncode==0, 'NET SHARE failed'
+    return co
```

Comment:
But `subprocess.run()` is new in Python 3.5.  Any problem hard-coding that  dependency in core Xonsh?  Not a problem for a xontrib module, but could be for core.